Frontend/ui-customer/src/Screens/CustomerUtils/ReportIssueChatScreen/Controller.purs

```diff
@@ -0,0 +1,475 @@
+                                             , data  { addAudioState { stateChanged = true } } }
+                  else
+                    pure $ UpdateState state { data  { recordedAudioUrl = Nothing } }
+  ]
+
+eval (RecordAudioModelAction RecordAudioModel.OnClickRestart) state =
+  continueWithCmd state [do
+    _ <- runEffectFn1 removeMediaPlayer ""
+    pure $ UpdateState state { data { recordAudioState { isRecording = false, recordingDone = false } } }
+  ]
+
+eval (RecordAudioModelAction RecordAudioModel.OnClickClose) state =
+  if state.data.recordAudioState.openAddAudioModel
+  then
+    continueWithCmd state { props { showRecordModel = false }, data { recordAudioState { recordedFile = Nothing, recordingDone = false, isRecording = false, openAddAudioModel = false, isUploading = false, timer = "00 : 00" } } } [do
+      _ <- runEffectFn1 removeMediaPlayer ""
+      _ <- runEffectFn1 stopAudioRecording ""
+      _ <- pure $ clearWaitingTimer state.props.timerId
+      pure $ UpdateState state { props { showRecordModel = false, showAudioModel = true, isPopupModelOpen = true }
+                   , data { recordAudioState { recordedFile = Nothing, recordingDone = false, isRecording = false, openAddAudioModel = false, isUploading = false, timer = "00 : 00" }, addAudioState { audioFile = Nothing, stateChanged = true } } }
+    ]
+  else
+    continue state { data  { recordAudioState { recordedFile = Nothing } }
+                   , props { isPopupModelOpen = false, showRecordModel = false } }
+
+eval (RecordAudioModelAction RecordAudioModel.BackPressed) state = do
+  continueWithCmd state [do
+    if state.props.isPopupModelOpen
+    then pure (RecordAudioModelAction RecordAudioModel.OnClickClose)
+    else pure BackPressed
+  ]
+
+eval (UpdateRecordModelPlayer url) state = do
+  continueWithCmd state { data { recordAudioState { recordedFile = Just url } } } [do
+    _ <- addMediaFile (getNewIDWithTag "recordedAudioViewUniqueOne") url (getNewIDWithTag "actionButtonRecord") "ny_ic_play_recorded_audio" "ny_ic_pause_recorded_audio" "-1"
+    pure NoAction
+  ]
+
+---------------------------------------------------- Chat View Model ----------------------------------------------------
+eval (ChatViewActionController (ChatView.SendSuggestion optionName)) state = do
+  let messages' = snoc state.data.chatConfig.messages (ChatView.makeChatComponent optionName "Customer" (getCurrentUTC ""))
+  case find (\x -> x.option == optionName) state.data.options of
+    Just selectedOption -> do
+      if (selectedOption.label == "CALL_DRIVER")then continue state { props { showCallDriverModel = true, isPopupModelOpen = true }, data{ selectedOption = Just selectedOption }} 
+        else  if (selectedOption.label == "REOPEN_TICKET") then exit $ ReopenIssue state { data{ selectedOption = Just selectedOption ,  options = [], chatConfig {suggestionsList = [], messages = messages'}} }
```

Comment:
create an enum for selectedOption.label, instead of using strings

Reply:
this values are coming from api

Reply:
if in future the variable changes from backend it has be changed at one single place, no need of changing it at all the places wherever it is used.